### PR TITLE
Convert ErgoDox EZ to Matrix Lite

### PR DESCRIPTION
**Pull request by drashna**
## Description

Like the title says.

Also added some config settings from the dactyl board, in prep to convert it. 

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodox_ez/matrix.c [hunk 1]
```diff
@@ -193,42 +124,47 @@ uint8_t matrix_key_count(void) {
...
+
+    // init on teensy
+    setPinInputHigh(F0);
+    setPinInputHigh(F1);
+    setPinInputHigh(F2);
+    setPinInputHigh(F3);
+    setPinInputHigh(F4);
+    setPinInputHigh(F5);
+    setPinInputHigh(F6);
+    setPinInputHigh(F7);
```
Comment (noroadsleft) [inline keyboards/ergodox_ez/matrix.c:138]: Why all the pins? Seems like F2 and F3 aren't used.

Comment (drashna) [reply keyboards/ergodox_ez/matrix.c:138]: Whoops. 
```suggestion
    // init on teensy
    setPinInputHigh(F0);
    setPinInputHigh(F1);
    setPinInputHigh(F4);
    setPinInputHigh(F5);
    setPinInputHigh(F6);
    setPinInputHigh(F7);
```

**Review by zvecr — APPROVED** [review]

File: keyboards/ergodox_ez/ergodox_ez.h [hunk 2]
```diff
@@ -4,7 +4,6 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include "i2c_master.h"
-#include <util/delay.h>
 
 #define CPU_PRESCALE(n) (CLKPR = 0x80, CLKPR = (n))
 #define CPU_16MHz       0x00
```
Comment (fauxpark) [inline keyboards/ergodox_ez/ergodox_ez.h:9]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — APPROVED** [review]
`__attribute__ ((weak))` :heavy_check_mark:

**Pull request merged**